### DISCUSSION TITLE
Consignment Question

### DISCUSSION THREAD

**Discussion by cymqqqq** [RGB-WG/rgb#219] [Q&A] [answered]
Description:

Now there is a `transfer` in the Consignment structure, the `transfer` is set to `false` when we issue a contract, and it is set to `true` when we finish an RGB transfer task. 

So my question is: if we have a new `issue` transition in the `inflation asset` schema, will there be a new consignment when we finish an `issue` transition? (Such as a new consignment generated, and set the `transfer` to false if the `issue` transition is done) Or is the consignment only related to the `transfer` transition?

**Comment by cymqqqq** [comment]
I noticed that in the consignment source code, there is `contains information about state transfer (true)`, so the `transfer` in the consignment means the state transition transfer but not the RGB token transfer, right?

**Comment by dr-orlovsky** [comment, marked as answer]
`Contract` (=`Consignment<false>`) and `Transfer` (=`Consignment<true>`) are two different forms of consignment which differentiate scenarios when the receiver will not change its own balance of any of the assets as a result of acceptance.

You create consignment for a specific receiver. If you do secondary issue and want to inform the public about the fact, this is `Contract`. If you issue to your wallet and need to be able to spend newly issued amount, this is `Transfer`.

**Reply by cymqqqq** [reply]
When I issue a new contract(it can be any kind of schema), it's public, and there will be a validated consignment(`Consignment<false>`) generated after the issue action, then will there must be a consignment(`Consignment<true>`) when I finished any kind of transition(except rgb20  token transfer) or extension?
